tests/compile/piecewise/test_simple.py

```diff
@@ -104,3 +109,37 @@ def test_simple_piecewise_compile(use_inductor):
             output = model(input)
         assert get_global_counter() == 2
         assert torch.allclose(output.cpu(), torch.tensor([19.0, 19.0]))
+
+
+@pytest.mark.parametrize("use_inductor", [True, False])
+@torch.inference_mode()
+def test_simple_piecewise_compile(use_inductor):
+    assert VLLM_USE_V1
+    _run_simple_model(
+        splitting_ops=["silly.attention"],
+        use_inductor_graph_partition=False,
+        use_inductor=use_inductor,
+        expected_num_piecewise_graphs_seen=5,  # 2 * num_layers + 1
+        expected_num_piecewise_capturable_graphs_seen=3,  # 1 + num_layers
+        expected_num_backend_compilations=
+        3,  # num_piecewise_capturable_graphs_seen
+    )
+
+
+@torch.inference_mode()
+@pytest.mark.parametrize("splitting_ops", [["silly.attention"], []])
+def test_simple_inductor_graph_partition(splitting_ops):
+    assert VLLM_USE_V1
+    _run_simple_model(
+        # inductor graph partition automatically resets splitting_ops
+        # to be an empty list
+        splitting_ops=splitting_ops,
+        use_inductor_graph_partition=True,
+        use_inductor=True,
+        expected_num_piecewise_graphs_seen=
+        1,  # since not splitting at fx graph level
+        expected_num_piecewise_capturable_graphs_seen=
+        1,  # since not splitting at fx graph level
+        expected_num_backend_compilations=
+        1,  # since not splitting at fx graph level
```

Comment:
How are we actually testing that the inductor graph splitting worked? Are we running cudagraph-unsafe ops in a test somewhere and checking that the output is correct?

Reply:
`test_simple_inductor_graph_partition` checks that, when `use_inductor_graph_partition=True`, we have 1 num_piecewise_graphs_seen and 1 num_backend_compilations. By contrast, `test_simple_piecewise_compile` checks that, when `use_inductor_graph_partition=True`, we have 5 num_piecewise_graphs_seen and 3 num_backend_compilations since we are splitting at fx graph level. For both tests, we assert that `num_cudagraph_captured=6`.

I also added `Tests` section in PR description.